core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -867,43 +867,98 @@ def _compute_permutation_importance(self, X, y, features: list, eval_metric=None
             transform_func=transform_func, transform_func_kwargs=transform_func_kwargs, silent=silent, **kwargs
         )
 
-    # TODO: PoC, refactor prior to v0.6 release
-    #  Add Documentation
-    #  Define when this should be called
-    #  Is it called during fit? Is it called after fit?
-    #  Can it be called as a post-fit operation on an already fit predictor on a per-model basis?
-    #  Does this call overwrite the existing model or make a new one?
-    def compile(self, path: str = None, verbose=True) -> str:
-        if path is None:
-            path = self.path
-        file_path = path + self.model_file_name
+    def can_compile(self, compiler_configs=None):
+        """
+        Verify whether the model can be compiled with the compiler configuration.
 
-        save_in_pkl = True
-        if self.model is not None:
-            self._compiler = self._get_compiler()
-            if self._compiler is not None:
-                self.model = self._compiler.compile(obj=self, path=path)
-                save_in_pkl = self._compiler.save_in_pkl
-        _model = self.model
-        if not save_in_pkl:
-            self.model = None
-        save_pkl.save(path=file_path, object=self, verbose=verbose)
-        self.model = _model
-        return path
+        Parameters
+        ----------
+        compiler_configs : dict, default=None
+            Model specific compiler options.
+            This can be useful to specify the compiler backend for a specific model,
+            e.g. {"RandomForest": {"compiler": "onnx"}}
+        """
+        if not self.is_fit():
+            return False
+        # Get model specific compiler options
+        # Model type can be described with either model type, or model name as string
+        if type(self) in compiler_configs:
+            configs = compiler_configs[type(self)]
+        elif self.name in compiler_configs:
+            configs = compiler_configs[self.name]
```

Comment:
Why are we allowing passing of other model compiler configs? This should be handled upstream, and `compiler_configs` should simply be `{"compiler": "onnx"}` in this context.

Reply:
This is kind of a model-specific compiler configration. For ensemble models, we could use a specific backend for a specific model. For instance, an ensemable model with RF and TORCH_NN, I think we could configure the compiler to optimize RF with onnx, while optimize TORCH_NN model with tvm, in order to maximize potential performance benefits. Of course, we can still use the top-level compiler config if model-specific compiler is not specified.

Reply:
```a model-specific compiler configration```

This shouldn't be available at the AbstractModel level, we only need the configuration for the model itself in this context, not others.

Reply:
Model-specific config logic has been moved outside of abstract_model.